components/back-compatile-on-input/index.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+export default class BackCompatibleOnInput extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onInput = this.onInput.bind( this );
+		this.onKeyUp = this.onKeyUp.bind( this );
+
+		this.state = {
+			isOnInputSupported: false,
+		};
+	}
+
+	onInput( event ) {
+		this.setState( {
+			onInputSupported: true,
+		} );
+		this.props.onInput( event );
+	}
+
+	onKeyUp( event ) {
+		this.props.onInput( event );
```

Comment:
For supported browsers, both the `onInput` and `onKeyUp` will be triggered on the first keypress. Is there a way we can cancel handling of the `onKeyUp` if we'd already done `onInput` ? Maybe an instance variable which is checked here (or state, if it's already reflected).

Reply:
It was improved, we now handle that case so we don't fire the first key two times.